# Exercise 3: Genome assembly quality assessment

## Assembly completeness - BUSCO

We like to use BUSCO to assess the completeness of an assembly. BUSCO (https://busco.ezlab.org/) uses a set of conserved genes that should be present in the species we are interested in. For instance, if you run it on a mammalian species, it tries to find 4104 genes. A high quality genome assembly should have the vast majority of these present and with exons in the correct order and orientation with regards to each other. In vertebrates, genes can span several 100 kbp, so if most genes are found complete, it is likely that the genome assembly is of high quality.

### Running BUSCO

First, let's do the assembly QC in a new folder, for example `~/assembly_qc`. Create this folder and `cd` into it.

Second, as the BUSCO analysis will take some time, we run it again inside a *screen*. It is advised to start a new screen for this

*NOTE* feel free to terminate any other screens you still may have by re-entering them and writing `exit`.

To get started with BUSCO, we need to run the following commands:

```
module load BUSCO/5.0.0-foss-2020a
```

To have a look at the available BUSCO gene sets for different species use:

`busco --list-datasets`

We will all run BUSCO using the insect dataset, **insecta_odb10**. If you want you could try a different one in addition, for instance for arthropods or eukaryotes.

To run BUSCO on the canu assembly use:

```
busco \
-c 2 \
-i /home/username/canu_assembly/canu_assembly.contigs.fasta \
-l insecta_odb10 \
-m genome \
-o canu_busco 1> canu_busco.out 2> canu_busco.err
```

For flye, use

```
busco \
-c 2 \
-i /home/username/flye_assembly/assembly.fasta \
-l insecta_odb10 \
-m genome \
-o flye_busco 1> flye_busco.out 2> flye_busco.err
```

As always, change `username` with your username and make sure the path to the assembly file is correct.

**NOTE** *If your assembly did not complete*, copy one of the finished assembly that we made available in the folders `/storage/BIOS3010/Genome_assembly/canu_assembly` or `/storage/BIOS3010/Genome_assembly/flye_assembly` to your home directory and run BUSCO on it.

In the command for running BUSCO:
* we use the backslash `\` so we can write the command over multiple lines, making it easier to read
* `-c 2` means use 2 compute threads (CPUs)
* `-i ` is the assembly file to use
* `-l` is the name if the gene set
* `-m genome` indicates we are testing a full genome (not only a set of gene or protein sequences)
* `-o` indicates the name of the folder BUSCO will write its output in
* `1> canu_busco.out 2> canu_busco.err` asks unix to save the messages from BUSCO in different files.

Exit the screen with `Ctrl + A + D` as before.

Check to see if BUSCO is running using `top` and checking the `.out` and `.log` files.

An example BUSCO report looks like this:

`C:99.9%[S:99.9%,D:0.0%],F:0.1%,M:-0.0%,n:781`

This means that 99.9 % were complete and 0.1 % were incomplete.

Look in the file called `busco_insecta.out`. It is the last 10 lines or so which are important, so you can for instance write:

`tail busco_insecta.out`

````diff
! What scores did you get?
! How do you interpret these scores? Is the assembly complete?
````

Pair up with someone who used the other program (canu or flye)
so you can compare BUSCO results.
For the question below, have a look inside the `canu_busco` or `flye_busco` folder and check the files in there.

````diff
! Compare the list of missing genes. Are the same genes missing in both assemblies?
````

## Assembly contiguity and correctness - QUAST

After doing a genome assembly, one of the questions is often about how correct and contiguous the assembly is. The term *contiguous* refers to how well the different seuqences have been put together. An assembly is mor contiguous the fewer contigs it has.

It is not straight-forward to evaluate correctness for many projects. However, for *Bombus campestris* we have a reference to compare to downloaded from https://www.ebi.ac.uk/ena/browser/view/GCA_905333015, located in the shared folder here:

`less /storage/BIOS3010/Genome_assembly/CAJOSK02.fasta`

This genome was generated using the following pipeline (from https://www.ebi.ac.uk/ena/browser/view/GCA_905333015):

> The assembly iyBomCamp1.2 is based on 83x PacBio data, 10X Genomics Chromium data, and Arima Hi-C data generated by the Darwin Tree of Life Project (https://www.darwintreeoflife.org/). The assembly process included the following sequence of steps: initial PacBio assembly generation with Hifiasm, retained haplotig separation with purge_dups, short-read polishing using FreeBayes-called variants from 10X Genomics Chromium reads aligned with LongRanger, and Hi-C based scaffolding with SALSA2. The mitochondrial genome was assembled using MitoHifi. Finally, the assembly was analysed and manually improved using gEVAL. Chromosome-scale scaffolds confirmed by the Hi-C data have been named in order of size.

This reference is the best available genome assembly for this species. We will assume it has no errors - but we can never be sure. It likely will have some errors, but probably not many.

We can use Quast (http://quast.sourceforge.net) to find many different metrics of our genome assembly such as N50, number of misassemblies relative to the reference, and number of genes. See more of the results given here: http://quast.sourceforge.net/docs/manual.html#sec3.1.

Note that we do not have *annotated* the genome yet. Annotation is the process that finds all the genes (and more). For our quast analysis, we will thus use the option `--gene-finding`.

To get started with Quast, we need to run the following commands:

```
module load QUAST/5.0.2-foss-2020a-Python-3.8.2
```

Work in your `assembly_qc` folder and use `screen` as before -
for example an existing screen.

To run Quast on the canu assembly use:

```
quast \
/home/username/canu_assembly/canu_assembly.contigs.fasta \
-r /storage/BIOS3010/Genome_assembly/CAJOSK02.fasta  \
--gene-finding \
--conserved-genes-finding \
-o canu_quast \
-t 2 \
1> canu_quast.out 2> canu_quast.err
```

For flye, use

```
quast \
/home/username/flye_assembly/assembly.fasta \
-r /storage/BIOS3010/Genome_assembly/CAJOSK02.fasta  \
--gene-finding \
--conserved-genes-finding \
-o flye_quast \
-t 2 \
1> flye_quast.out 2> flye_quast.err
```

As always, change `username` with your username and make sure the path to the assembly file is correct.

Here:
* we use the path to the assembly fasta file as first argument
* `-r` indicates the path to the reference genome that Quast uses for comparison
* `--gene-finding` is explained above

--conserved-genes-finding \
* `-o` indicates the name of the folder Quast will write its output in
* `-t 2` means use 2 compute threads (CPUs)


If you want to run it on your own assembly, make sure the paths are correct.

Exit the screen with `Ctrl + A + D` as before.

Check to see if Quast is running using `top` and checking the `.out` file.


Quast creates a subfolder with many different files. Take a look around in it. Then, look at the file called

`less quast_results/latest/report.txt`

````diff
! How many contigs longer than 50000 bp does your assembly consist of?
! What is the N50 length?
! How many misassemblies are there?
````

There is also a PDF file which you can download to your computer to look at.

I have uploaded the canu and flye quast reports to the github folder so you can compare the assembly programs.

````diff
! What are the main differences between flye and canu? (N50, misassemblies, longest contig…)
! Which assembly is more contiguous?
````

Navigate to where the reference assembly is located https://www.ebi.ac.uk/ena/browser/view/GCA_905333015

<img width="1263" alt="Screenshot 2021-05-01 at 13 47 44" src="https://user-images.githubusercontent.com/46928237/116781542-d1aac700-aa83-11eb-8b60-a55f5abf23a0.png">

Click on the “Assembly Statistics” on the right hand side.

````diff
! How does the reference genome assembly compare to your assembly in terms of contiguity?
````

You can also download the sequence report (click “Sequence Report” on the right hand side.

````diff
! How many chromosomes does the assembly contain?
````
